src/query/procedure/module.cpp

```diff
@@ -313,13 +308,386 @@ void RegisterMgTransformations(const std::map<std::string, std::unique_ptr<Modul
+}
+}  // namespace
+
+void RegisterMgGetModuleFiles(ModuleRegistry *module_registry, BuiltinModule *module) {
+  auto get_module_files_cb = [module_registry](mgp_list * /*args*/, mgp_graph * /*unused*/, mgp_result *result,
+                                               mgp_memory *memory) {
+    for (const auto &module_directory : module_registry->GetModulesDirectory()) {
+      for (const auto &dir_entry : std::filesystem::recursive_directory_iterator(module_directory)) {
+        if (dir_entry.is_regular_file() && IsAllowedExtension(dir_entry.path().extension())) {
+          mgp_result_record *record{nullptr};
+          {
+            const auto success = TryOrSetError([&] { return mgp_result_new_record(result, &record); }, result);
+            if (!success) {
+              return;
+            }
+          }
+          const auto path_string = GetPathString(dir_entry);
+          const auto is_editable = IsFileEditable(dir_entry);
+
+          const auto path_value = GetStringValueOrSetError(path_string.c_str(), memory, result);
+          if (!path_value) {
+            return;
+          }
+
+          MgpUniquePtr<mgp_value> is_editable_value{nullptr, mgp_value_destroy};
+          {
+            const auto success = TryOrSetError(
+                [&] { return CreateMgpObject(is_editable_value, mgp_value_make_bool, is_editable, memory); }, result);
+            if (!success) {
+              return;
+            }
+          }
+
+          if (!InsertResultOrSetError(result, record, "path", path_value.get())) {
+            return;
+          }
+
+          if (!InsertResultOrSetError(result, record, "is_editable", is_editable_value.get())) {
+            return;
+          }
+        }
+      }
+    }
+  };
+
+  mgp_proc get_module_files("get_module_files", get_module_files_cb, utils::NewDeleteResource(),
+                            {.required_privilege = AuthQuery::Privilege::MODULE_READ});
+  MG_ASSERT(mgp_proc_add_result(&get_module_files, "path", Call<mgp_type *>(mgp_type_string)) == MGP_ERROR_NO_ERROR);
+  MG_ASSERT(mgp_proc_add_result(&get_module_files, "is_editable", Call<mgp_type *>(mgp_type_bool)) ==
+            MGP_ERROR_NO_ERROR);
+  module->AddProcedure("get_module_files", std::move(get_module_files));
+}
+
+void RegisterMgGetModuleFile(ModuleRegistry *module_registry, BuiltinModule *module) {
+  auto get_module_file_cb = [module_registry](mgp_list *args, mgp_graph * /*unused*/, mgp_result *result,
+                                              mgp_memory *memory) {
+    MG_ASSERT(Call<size_t>(mgp_list_size, args) == 1U, "Should have been type checked already");
+    auto *arg = Call<mgp_value *>(mgp_list_at, args, 0);
+    MG_ASSERT(CallBool(mgp_value_is_string, arg), "Should have been type checked already");
+    const char *path_str{nullptr};
+    {
+      const auto success = TryOrSetError([&] { return mgp_value_get_string(arg, &path_str); }, result);
+      if (!success) {
+        return;
+      }
+    }
```

Comment:
```suggestion
      if (const auto success = TryOrSetError([&] { return mgp_value_get_string(arg, &path_str); }, result); !success) {
        return;
      }
```

Reply:
oh wow, How did I miss this and so many times, great catch!
I can just do `if (!TryOrSetError..)`